RATEST-242: RefApp 2.x Patient condition workflow test

mherman22 opened the pull request:
**Ticket ID**
https://issues.openmrs.org/browse/RATEST-242

**reference**
https://github.com/openmrs/openmrs-contrib-qaframework/commit/247b49d8974a89f5f3e3d6c2c162639f3fec6349

Work Done

1. Added visibility Elements to the editInactive element and delete inactive element.
2. Otherwise this is a continuation of https://github.com/openmrs/openmrs-contrib-qaframework/pull/174

cc: @kdaud @sherrif10

kdaud commented:
@mherman22 is the commit reflecting what is taking place in the PR?

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -171,33 +181,36 @@ public void successiveInactiveEdition() {
+		String name = conditionsPage.getFirstConditionName();
 		if (StringUtils.isNotBlank(name)) {
-			assertNull(driver.findElement(By.linkText(name)));
+			assertNull(driver.findElement(By.xpath(name)));
```
isnt there anything specific than asserting an element.

mherman22 replied:
not been able to find one. would you care to suggest?

kdaud replied:
@sherrif10 do you have any response on the caption [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/230/files#r751346120)?

sherrif10 replied:
@kdaud  am okay with that approach after inspecting it.

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -25,27 +26,30 @@
 
 public class ConditionsSteps extends Steps {
-	
-	private ConditionsPage conditionsPage;
```
Seems you are changing the format from the former formatting , was it a recommendation ?

mherman22 replied:
Yes it was a recommendation to have the test create its own patient

kdaud commented:
@mherman22 is this PR adding visibility element to Patient condition workflow test as the commit title starts?

mherman22 commented:
@kdaud refering to your question https://github.com/openmrs/openmrs-contrib-qaframework/pull/230?notification_referrer_id=NT_kwDOAs7_ibMyNjk5MDk4NDkzOjQ3MTIwMjY1&notifications_query=is%3Aunread#issuecomment-972778069 , i believe i did so. i changed the cssSelectors of the delete buttons and replaced them with the new selectors as per [30](https://github.com/openmrs/openmrs-contrib-qaframework/blob/fc79886464d38b0168b18abe3002d580acd6cee6/src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java#L30) and [31](https://github.com/openmrs/openmrs-contrib-qaframework/blob/fc79886464d38b0168b18abe3002d580acd6cee6/src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java#L31). I then added `waiter.until(ExpectedConditions.elementToBeClickable` to those elements which were not visible. i didnt use visibilityOfElementLocated because all it does is check whether the element is visible on the DOM. i used `elementToBeClickable` because it is an expectation for checking an element is visible and enabled such that you can click it.

referred to https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/ui/ExpectedConditions.html

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,66 @@
+    # User story: Adding an active condition
+    And User enters active condition
+    And User clicks save
+    
```
was it an oversight to leave out this empty line?
```suggestion
```

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,66 @@
+    # User story: Adding an active condition
+    And User enters inactive condition
+    And User clicks save
+    
```
```suggestion
```

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,66 @@
+  Scenario: Editing First Active condition
+    And User enters active condition
+    And User clicks save
+    
```
```suggestion
```

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,66 @@
+  Scenario: Editing First Inactive condition
+    And User enters inactive condition
+    And User clicks save
+    
```
```suggestion
```

kdaud reviewed (approved):
@sherrif10 what do you say about this PR?

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -25,27 +26,30 @@
+    }
+    
 	@Given("User clicks on Conditions from Patient dashboard")
 	public void launchManageConditions() {
```
Fix the identation to be same . You have two factor identation format, cross check and confirm especially the before and after hook and other methods definition.

mherman22 replied:
@sherrif10 i have fixed the indention in the commit https://github.com/openmrs/openmrs-contrib-qaframework/pull/230/commits/34fc2eba733c5a1ecf2e395ef33ec2d6cbd11cb0

sherrif10 commented:
We can make the title as `RATEST-242: Adding Visibility Element to refapp-2.x Patient conditions workflow`

sherrif10 reviewed (approved):
Thanks @mherman22   . LGTM  cc @dkayiwa  @kdaud this seems ripen for entering into master
